tests/test_calc_error.py

```diff
@@ -0,0 +1,459 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+import numpy as np
+import unittest
+
+from prog_models import *
+from prog_models.models import *
+
+class TestCalcError(unittest.TestCase):
+    """
+    Main Testing class for calc_error.
+
+    Validating that values are correctly being passed into the new calc_error calls and that we are receiving expected results!
+    """
+    def test_calc_error(self):
+        # Note, lowering time steps or increasing simulate threshold may cause this model to not run (takes too long)
+        m = BatteryElectroChemEOD()
+
+        options = {
+            'save_freq': 200, # Frequency at which results are saved
+            'dt': 1, # Time step
+        }
+
+        def future_loading(t, x=None):
+            if (t < 600):
+                i = 2
+            elif (t < 900):
+                i = 1
+            elif (t < 1800):
+                i = 4
+            elif (t < 3000):
+                i = 2
+            else:
+                i = 3
+            return m.InputContainer({'i': i})
+    
+        simulated_results = m.simulate_to(2000, future_loading, **options)
+
+        # Running calc_error before setting incorrect parameters
+        m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)
+
+        with self.assertWarns(UserWarning) as cm:
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt = 1, stability_tol = 10)
+        self.assertEqual(
+            "Configurable cutoff must be some float value in the domain (0, 1]."  
+            " Received 10. Resetting value to 0.95",
+            str(cm.warning)
+        )
+
+        # Initializing parameters to very erroneous values       
+        m.parameters['qMax'] = 4000
+        keys = ['qMax']
+
+        # Before running estimate_params
+        with self.assertRaises(ValueError):
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)
+
+        m.estimate_params([(simulated_results.times, simulated_results.inputs, simulated_results.outputs)], keys, dt=0.5)
+
+        # After running estimate_params. Note that this would not change the outcome of the result
+        with self.assertRaises(ValueError):
+            m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)    
+
+        # Running through various dt values that could work
```

Comment:
Can we make this comment more descriptive? I'm not sure what the point of this test is